services/ui_backend_service/tests/integration_tests/tasks_test.py

```diff
@@ -494,17 +497,32 @@ async def create_ok_artifact_for_task(db, task, attempt=0):
     return _task
 
 
-async def create_task(db, step=None, status="running", task_id=None, task_name=None, last_heartbeat_ts=None):
```

Comment:
I forked this.  Previously `step` is an optional arg.  If no `step` is provided, `create_task` created its own stack of flow, run, step because creating the task.  We did not return any of the created items, and it would have been awkward to do so (return a tuple?).

Instead I separated into `create_task_with_step` - where the caller is responsible for preparing ancestor objects.  It then becomes clearer exactly when and how we apply ancestral run tags on a created task.